include/lsst/afw/coord/Observatory.h

```diff
@@ -85,10 +85,18 @@ class Observatory final {
     std::string toString() const;
 
     bool operator==(Observatory const& rhs) const noexcept {
-        auto deltaLongitude = (_latitude - rhs.getLatitude()).wrapCtr();
-        auto deltaLatitude = (_longitude - rhs.getLongitude()).wrapCtr();
+        auto deltaLongitude =
+                std::isfinite(_latitude.asRadians()) && std::isfinite(rhs.getLatitude().asRadians())
+                        ? (_latitude - rhs.getLatitude()).wrapCtr()
+                        : 0.0 * lsst::geom::degrees;
+        auto deltaLatitude =
+                std::isfinite(_longitude.asRadians()) && std::isfinite(rhs.getLongitude().asRadians())
+                        ? (_longitude - rhs.getLongitude()).wrapCtr()
+                        : 0.0 * lsst::geom::degrees;
```

Comment:
This logic is very surprising to me. When comparing two Observatories, one with `+Inf` latitude and longitude is equal to _all_ other Observatories? I would have expected Inf to only equal Inf (maybe ignoring sign, I guess).
```